### PR TITLE
Update for Flarum beta 8

### PR DESCRIPTION
**Pull request by dsevillamartin**
**Changes made:**
- Updated `bootstrap.php` to utilize new Extenders
  - Renamed to `extend.php`
- Updated PHP namespaces
- Updated icon to FA 5
- Updated `UrlGenerator` usage
  - Route collections, added `->to('forum')`
- Updated `RedirectMiddleware` with new middleware interface
- Updated JS structure
  - Switched to `webpack`
  - Updated modal form to JSX
- Removed PHP constraint in `composer.json`
  - Flarum beta 8 requires >=7.1

**Note**: Still using `ConfigureMiddleware`, which is deprecated in beta 8. Only alternative as far as I can see is using Laravel container events, aka `app->resolving('flarum.forum.middleware', function...)`
**Tests**: I have updated the tests for beta 8 as well, however I was unable to test them because of the beta 8 constraint. Travis is gonna fail as well, so I guess we'll see when beta 8 comes out.

image
image

### CODE REVIEW

**Comment by clarkwinkelmann** [comment]
Thanks a lot ! I'll test once beta8 is out :)

File: composer.json [hunk 1]
```diff
@@ -21,8 +21,7 @@
         "source": "https://github.com/migratetoflarum/vbulletin-redirects"
     },
     "require": {
-        "php": ">=7.1",
-        "flarum/core": "^0.1.0-beta.7",
+        "flarum/core": "^0.1.0-beta.8",
         "flarum/flarum-ext-tags": "^0.1.0-beta.8"
```
Comment (luceos) [inline composer.json]:
```
 ~/Sites/foo.com  master  composer why flarum/flarum-ext-tags                                                                                        11:43:49  ✓
migratetoflarum/vbulletin-redirects  0.1.2  requires  flarum/flarum-ext-tags (^0.1.0-beta.8)
```

The package has been renamed, please use flarum/tags.

**Comment by luceos** [comment]
@clarkwinkelmann would it be possible to merge this so we can patch the client site ahead of beta 8 release?

**Comment by clarkwinkelmann** [comment]
@luceos you want it merged to master or merged + tagged ? I don't understand why you need that ? :thinking: 

I still wanted to do some checks on this PR before merging.

**Comment by luceos** [comment]
I'd prefer having it in dev-master so I can move forward with getting a staging environment up. I'll patch any issues with this package if I find it.

**Comment by clarkwinkelmann** [comment]
Here we go, merged to master.

There were indeed still a few things to fix, which I did.

Tests still won't run out of the box, but if you set both dependencies to `dev-master` everything runs green now :wink:

**Comment by luceos** [comment]
I noticed, the forum is back online. Thank you 🙇

**Pull request merged**